### PR TITLE
feat: implemented OSS to Cloud hints for OSS

### PR DESCRIPTION
**Pull request by Toxer10**
image
image

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -91,27 +100,69 @@ const App: React.FC<any> = () => {
...
-      <Route path="*" element={<NotFound />} />
-    </Routes>
-    {isFullScreenLogOutput ? <LogOutputHeader logOutput={logOutput} isFullScreen /> : null}
-    <CSSTransition nodeRef={logRef} in={isFullScreenLogOutput} timeout={1000} classNames="full-screen-log-output" unmountOnExit>
-      <FullScreenLogOutput ref={logRef} logOutput={logOutput} />
-    </CSSTransition>
-  </Suspense>;
+  return (
+    <Suspense fallback={<Loading />}>
+      {!isTestkubeCloudLaunchBannerHidden && (
```
Comment (rangoo94) [inline src/App.tsx:104]: It's better to use `condition ? <Component /> : null` instead of `condition && <Component />`, to avoid printing `condition` in case it's falsy.

File: src/components/organisms/Sider/Sider.styled.tsx [hunk 2]
```diff
@@ -84,16 +84,18 @@ export const StyledSiderLink = styled(SiderLink)`
...
   justify-content: center;
   align-items: center;
 
   transition: 0.3s;
 
   svg {
-    width: 20px;
-    height: 20px;
+    width: ${({$size}) => $size}px;
+    width: ${({$size}) => $size}px;
```
Comment (rangoo94) [inline src/components/organisms/Sider/Sider.styled.tsx:98]: 1. `width`/`height`, not `width`/`width`
2. Why is that needed? Maybe `1em` will be better?
3. `$size` may be `undefined`

File: src/App.tsx (hunk 1 above)
Comment (Toxer10) [reply src/App.tsx:104]: I'm thinking about this all the time. Let's use `.. ? .. : ..`, agreed ✅

File: src/components/organisms/Sider/Sider.tsx [hunk 3]
```diff
@@ -102,9 +110,39 @@ const Sider: React.FC = () => {
       icon: 'cog',
       onClick: () => navigate('/settings'),
     },
-    {icon: 'github', onClick: () => window.open(externalLinks.github)},
-    {icon: 'documentation', onClick: () => window.open(externalLinks.documentation)},
-    {icon: 'discord', onClick: () => window.open(externalLinks.discord)},
+    {
```
[the remark this answers is not in the document]
Comment (Toxer10) [reply src/components/organisms/Sider/Sider.tsx:113]: For both: yes, it's a part of ticket. [Here](https://github.com/kubeshop/testkube-cloud/issues/409), the first bullet

File: src/components/organisms/Sider/Sider.styled.tsx (hunk 2 above)
Comment (Toxer10) [reply src/components/organisms/Sider/Sider.styled.tsx:98]: 1. Agree.
2. To force `width` and `height` of icons, we kinda need it :)
3. `$size` is 20 by default.

Comment (rangoo94) [reply src/components/organisms/Sider/Sider.styled.tsx]: 2. I meant basing on font size. It's common way for icons.
```
font-size: 20px;
width: 1em;
height: 1em
```

3. Looks like it's not - `$size?: number;`

File: .stylelintrc.json [hunk 4]
```diff
@@ -31,6 +31,7 @@
     "selector-pseudo-element-no-unknown": true,
     "selector-type-no-unknown": true,
     "string-no-newline": true,
-    "unit-no-unknown": true
+    "unit-no-unknown": true,
+    "selector-class-pattern": "^([a-zA-Z][a-zA-Z0-9]*)(-[a-zA-Z0-9]+)*$"
```
Comment (rangoo94) [inline .stylelintrc.json:35]: Probably it will be better to leave it as it was, and just add `/* stylelint-disable selector-class-pattern */` in this single place. Optionally, disable it all.

File: src/App.tsx [hunk 5]
```diff
@@ -49,6 +54,10 @@ const App: React.FC<any> = () => {
 
   const [isEndpointModalVisible, setEndpointModalState] = useState(false);
 
+  const update = useUpdate();
+
+  const isTestkubeCloudLaunchBannerHidden = localStorage.getItem(config.isTestkubeCloudLaunchBannerHidden);
```
Comment (rangoo94) [inline src/App.tsx:59]: Optionally:
```suggestion
  const isTestkubeCloudLaunchBannerHidden = localStorage.getItem(config.isTestkubeCloudLaunchBannerHidden) === 'true';
```

File: src/components/molecules/MessagePanel/MessagePanel.styled.tsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    a {
+      color: ${Colors.rose200};
+    }
+  }
+
+  &.warning {
+    a {
+      color: ${Colors.amber200};
+    }
+  }
```
Comment (rangoo94) [inline src/components/molecules/MessagePanel/MessagePanel.styled.tsx:71]: Given it's the same color as for the description:
```suggestion
  a {
    font-weight: 500;
    text-decoration: underline;
    color: inherit;
  }
```

File: src/components/organisms/Sider/Sider.styled.tsx [hunk 7]
```diff
@@ -114,3 +116,33 @@ export const StyledOtherItem = styled.div`
...
+    }
+
+    svg {
+      path {
+        fill: ${Colors.whitePure};
+
+        transition: 0.3s;
+      }
+    }
+  }
```
Comment (rangoo94) [inline src/components/organisms/Sider/Sider.styled.tsx:147]: I'm not sure if that's the case here, but if everything in `a` has to be same color:
```suggestion
  &:hover {
    color: ${Colors.whitePure};

    .testkube-text {
      color: inherit;
    }

    svg {
      path {
        fill: currentColor;
        transition: 0.3s;
      }
    }
  }
```

File: src/components/organisms/Sider/Sider.styled.tsx [hunk 8]
```diff
@@ -114,3 +116,33 @@ export const StyledOtherItem = styled.div`
...
+  cursor: pointer;
+  transition: 0.3s;
+
+  a {
+    flex: 1;
+
+    padding: 14px;
+  }
+
+  &:hover {
```
Comment (rangoo94) [inline src/components/organisms/Sider/Sider.styled.tsx:135]: It should be `:hover` on `a` (link), not on the list element, as you may interact with the link, not `li`.

File: src/components/organisms/Sider/Sider.tsx [hunk 9]
```diff
@@ -101,10 +108,53 @@ const Sider: React.FC = () => {
...
+              </DropdownListItem>
+            </>
+          }
+          trigger={['hover']}
+        >
+          <QuestionCircleOutlined
+            style={{
+              fontSize: 20,
+            }}
+          />
```
Comment (rangoo94) [inline src/components/organisms/Sider/Sider.tsx:148]: Optionally, as it's pretty short:
```suggestion
          <QuestionCircleOutlined style={{  fontSize: 20 }} />
```

File: src/styles/globalStyles.ts [hunk 10]
```diff
@@ -340,6 +340,14 @@ export const GlobalStyle = createGlobalStyle`
       color: ${Colors.slate500};
     }
   }
+  
+  /* Popover */
+
+  .ant-popover-placement-rightBottom {
+    .ant-popover-arrow  {
+      bottom: 4px;
+    }
```
Comment (rangoo94) [inline src/styles/globalStyles.ts:349]: Should it apply to all popovers?

File: src/styles/globalStyles.ts [hunk 11]
```diff
@@ -340,6 +340,14 @@ export const GlobalStyle = createGlobalStyle`
       color: ${Colors.slate500};
     }
   }
+  
+  /* Popover */
+
+  .ant-popover-placement-rightBottom {
+    .ant-popover-arrow  {
```
Comment (rangoo94) [inline src/styles/globalStyles.ts:347]:
```suggestion
    .ant-popover-arrow {
```

File: src/components/molecules/MessagePanel/MessagePanel.styled.tsx (hunk 6 above)
Comment (Toxer10) [reply src/components/molecules/MessagePanel/MessagePanel.styled.tsx:71]: Agree

File: src/components/organisms/Sider/Sider.styled.tsx (hunk 7 above)
Comment (Toxer10) [reply src/components/organisms/Sider/Sider.styled.tsx:147]: Checked, does not work as expected

https://github.com/kubeshop/testkube-dashboard/assets/25991946/5299b8aa-4238-4e8f-96a6-779133c8b80b

File: src/styles/globalStyles.ts (hunk 10 above)
Comment (Toxer10) [reply src/styles/globalStyles.ts:350]: I found a better way, reworked it
